### PR TITLE
nogo: ignore generated source files

### PR DESCRIPTION
**Pull request by sluongng**
Under GoCompilePkg actions, there are several ways that source files
could have been modified / generated that is out of end users
control:

- go_test would generate an '_empty.go' file for the missing
  internal/external package to compile it.

- coverage instrumentation would rewrite test source files with a
  wrapper source file that collects coverage data into a determined
  variable.

- CGO usage would transform the original source files into several
  output Go and C source files with `go tool cgo ...`

Implement a mechanism to ignore the generated source files for the first
2 cases and instruct static analyzers to run over the original source
files wherever applicable.  This provides a better default for rules_go
users as static analysis errors over these generated files are outside
of their control and thus, could not be fixed.

Add a note regarding the CGO case for future investigations.

Replacement of #2996
Fixes https://github.com/bazelbuild/rules_go/pull/2984

### CODE REVIEW

File: .bazelci/presubmit.yml [hunk 1]
```diff
@@ -295,6 +295,7 @@ tasks:
     - "-//tests/core/go_test:data_test"
     - "-//tests/core/go_test:pwd_test"
     - "-//tests/core/nogo/coverage:coverage_test"
+    - "-//tests/core/nogo/coverage:gen_code_test"
```
Comment (sluongng) [inline .bazelci/presubmit.yml:298]: See https://github.com/bazelbuild/rules_go/issues/2518 for details

**Comment by sluongng** [comment]
The reason why I skip on doing CGO is:
- It's a bit more complex to dive into how `cgo2()` work
- I have yet to encounter a CGO generated code conflicting with static analyzers myself.

Would love if somebody could address that in the future as I believe it would make the code a lot 'cleaner'.

File: tests/core/nogo/coverage/README.rst [hunk 2]
```diff
@@ -19,10 +19,5 @@ Verifies `#2146`_.
...
 gen_code_test
 -------------
-Checks how `nogo`_ would interact with source code that was generated as part of
-rules_go's coverage implementation.  Currently `nogo`_ would be run over these
-generated source code, which is not desirable as end-user have very little control
-over how these code were generated.
-
-In a future version, we shall flip this behavior so that `nogo`_ would not run
-over source files that users have no control over.
+Checks how `nogo`_ shoudl not interact with source code that was generated as part of
```
Comment (fmeum) [inline tests/core/nogo/coverage/README.rst:22]:
```suggestion
Checks how `nogo`_ should not run on source code that was generated as part of
```

**Comment by fmeum** [comment]
Not dealing with CGo code until there is a clear need to do so seems fine to me.

**Comment by fmeum** [comment]
@aiuto Sorry to bother you with this, but is there a way to sign the Google CLA for a single GitHub account with two Google accounts? 

My last name changed and I now maintain two distinct Google accounts linked to my old and new name. I have signed the CLA with the old one and now can't add my new email to it because myaccount.google.com/alternateemail doesn't allow entering the address of another Google account. Do you see a way to unblock my commits under the new name?

**Review by fmeum — APPROVED** [review]

**Review by fmeum — APPROVED** [review]

**Pull request merged**